### ISSUE TITLE
Logging objects that have DOMelements etc. within causing huge logs + throwing electron error 

### ISSUE THREAD

**Issue by delewis13** [megahertz/electron-log#222]
Hello and thanks for this great library!

Was in the process of changing the majority of my console.logs in renderer to output to log file when I noticed that a subset of the log statements cause serious [sometimes fatal] lag, and in some cases throw the following error: 

```
Blocked a frame with origin "http://localhost:9080" from accessing a cross-origin frame.
Call Stack
 eval
  node_modules/electron-log/src/transform/object.js:128:22
 toStringFunction
  node_modules/electron-log/src/transform/object.js:123:31
 eval
  node_modules/electron-log/src/transform/index.js:47:14
 Object.transform
  node_modules/electron-log/src/transform/index.js:45:23
 transport
  node_modules/electron-log/src/transports/file/index.js:60:29
 runTransport
  node_modules/electron-log/src/log.js:44:5
 runTransports
  node_modules/electron-log/src/log.js:27:7
```

I'm sure you are already familiar with this, but it seems to be happening primarily when my logs contain a deeply nested object [e.g. logging out a domNode]. I'm guessing that's because in the console there is some lazy-loading going on, while when you log to file the whole thing needs to be evaluated.

As such, one obvious solution [which is what i've done], is just not log those things, as I probably shouldn't have been anyway. 

Am wondering however if there is some other solution whereby you can limit the depth to which electron-log travels the 
object? Seems like it's a little too easy for me to crash my program just by logging something which I shouldn't have :S

Cheers, hope it's suitable to post here. Can move to stackoverflow if you think that's the better forum for this kind of thing?

**Comment by megahertz** [comment]
Hi, I haven't test how DOM nodes are logged in the last version of Electron, I'm going to play with it a bit later. As a workaround (when you need these elements in logs), you can play with `log.transports.file.inspectOptions.depth` option. The default value 5 may be too large for DOM nodes.

**Comment by delewis13** [comment]
> Hi, I haven't test how DOM nodes are logged in the last version of Electron, I'm going to play with it a bit later. As a workaround (when you need these elements in logs), you can play with `log.transports.file.inspectOptions.depth` option. The default value 5 may be too large for DOM nodes.

Thanks for the response :) Hmm unfortunately this didn't do the trick - guess it's not an issue with depth after all. 

I tried to re-create a somewhat-minimal example [using react + typescript + electron-webpack], but was not able to reproduce the error. Must be something funky in my project setup that i'm missing... While doing so I realised I was wrong about DOMNodes - it's actually events that cause the issue not DOM nodes [e.g. logging an event attached to an onClick handler of a button is enough to cause it]

Some observations which may or may not be useful. The attributes of `event.view.global` object appear to be the culprits, in particular: `parent` [which has global as a sub attribute], `top`, `frames`, `self`, `window`, `document`, `global` and `module`. Of these, I think all except for `module` has a recursive reference to global.

Unfortunately I don't have too much time right now, but my jobs next release is done, i'll have another crack at the minimal example. 

Cheers :)
